src/ui/public/kuery/lib/node_types/compound.js

```diff
@@ -0,0 +1,56 @@
+import {
+  toElasticsearchQuery as nodeToElasticsearchQuery,
+  toKueryExpression as nodeToKueryExpression,
+} from '../ast';
+
+export function buildNode({ children, operation }) {
```

Comment:
@weltenwort don't waste too much time worrying about the implementation of these functions. While I plan to keep the `node_type` interface the same, the implementation will change along with the changes to the AST I was showing you. During our sync we could discuss whether it makes sense to bring those changes into this PR.